charts/tekton-apps/templates/eventlistener.yaml

```diff
@@ -14,16 +14,25 @@ spec:
   # ========================================================================
   # {{ $project.project | upper }} PROJECT
   {{- range $component := $project.components }}
+
+  {{- /* initialize `projectEnvironment` variable with either `$project.environment` if it exists or with `$.Values.environment` otherwise */}}
+  {{ $projectEnvironment := ternary $project.environment $.Values.environment (hasKey $project "environment") -}}
+
+  {{- /* initialize `gitBranchPrefixes` variable with either `$component.eventlistener.gitWebhookBranches` list if it exists or `$.Values.gitBranchPrefixes` otherwise */}}
+  {{ $gitBranchPrefixes := ternary $component.eventlistener.gitWebhookBranches $.Values.gitBranchPrefixes (hasKey $component.eventlistener "gitWebhookBranches") -}}
+
   # ========================================================================
   # 🐤 {{ $project.project | upper }}/{{ $component.name | upper }} COMPONENT
   # ========================================================================
+
   {{- $data := dict "project"           $project
                     "component"         $component
-                    "environment"       $.Values.environment
+                    "environment"       $projectEnvironment
                     "eventlistener"     $.Values.eventlistener
-                    "gitBranchPrefixes" $.Values.gitBranchPrefixes }}
+                    "gitBranchPrefixes" $gitBranchPrefixes }}
   {{- include "tekton-apps.eventlistener.trigger" $data | nindent 4 }}
-  {{- end }} # range component
-  {{- end }} # if project.enabled
-  {{- end }} # range
-  {{- end }} # if apps
+
+{{- end }} {{- /* range component */}}
```

Comment:
I personally don't like commenting this way in helm chart as it makes code clogged.
I know that if we don't do that then these commends ends up in YAML manifest in k8s but I think it's ok. 

HELM code is not always easy to read. So the less parenthesis we have - the better. 
Thoughts?

Reply:
let's change though templates/basic -> templates/general (or templates/common).

Reply:
Ok with folder renaming, but don't agree with comments. I think it is not good and generates a mess..

Reply:
Why? Reading HELM chart with a lot of parenthesis is a pain. I'd rather prefer # in the YAML manifest that you see in kubernetes (and pay no attention) than to see a lot more parenthesis inside the HELM chart that will require our attention.

Reply:
I Created the vote in the slack channel. Let's see

Reply:
done with folder rename